extensions/websocket/README.md

```diff
@@ -0,0 +1,436 @@
+constructor.
+
+With an express instance:
+```typescript
+import { WebsocketBindings, WebsocketComponent } from '@loopback/websocket';
+
+...
+
+this.component(WebsocketComponent);
+
+const expressApp = express();
+this.bind(WebsocketBindings.REQUEST_LISTENER).to(expressApp);
+
+this.wsServer = this.getSync(WebsocketBindings.SERVER);
+```
+
+With a RestApplication application
+```typescript
+import { WebsocketBindings, WebsocketComponent } from '@loopback/websocket';
+
+...
+
+this.component(WebsocketComponent);
+// this.requestHandler is a property of RestApplication
+this.bind(WebsocketBindings.REQUEST_LISTENER).to(this.requestHandler);
+
+this.wsServer = this.getSync(WebsocketBindings.SERVER);
+```
+
+#### Configuration
+The configuration for the `WebsocketServer` can by send via options of
+the application constructor
+```typescript
+const app = new MyApplication({
+  websocket: {
+    host: '127.0.0.1',
+    port: 0,
+  },
+});
+```
+
+### Register controllers
+
+To register a controller use `route` method of `WebsocketServer` instance.
+```typescript
+export class ChatController {}
+
+...
+
+// To root namespace
+this.wsServer.route(ChatController);
+// To specific namespace
+this.wsServer.route(ChatController, '/ws/chat');
+this.wsServer.route(ChatController, /^\/ws\/chat$/);
+```
+
+Controller can be decorator with `@ws.controller()` to specify the namespace
+to handle.
+```typescript
+import { ws } from '@loopback/websocket';
+
+@ws.controller('/ws/chat')
+export class ChatController {}
+
+...
+
+this.wsServer.route(ChatController);
+```
+
+### Subscribe method to events
+To subscribe a controller's method to an event use decorators: 
+
+```typescript
+import { ws } from '@loopback/websocket';
+
+export class ChatController {
+  // Subscribe to connection
+  @ws.connect()
+  connect() {}
+  
+  // Subscribe to event by string
+  @ws.subscribe('chat message')
+  handleChatMessage() {}
+
+  // Subscribe to event by regex
+  @ws.subscribe(/.+/)
+  logMessage() {}
+
+  // Subscribe to disconnection
+  @ws.disconnect()
+  disconnect() {}
+}
+```
+
+### Injection of elements
+Different objects can be injected into controller's methods subscribed to event
+of socket.
+
+#### Inject connection socket
+Socket can be injected in methods decorated with `@ws.connect`, `@ws.subscribe` or
+`@ws.disconnect`. Also, you can inject it in the controller constructor.
+```typescript
+import { Socket } from 'socket.io';
+import { ws } from '@loopback/websocket';
+
+export class ChatController {
+  constructor(@ws.socket() socket: Socket){
+    console.log('controller instance created for: %s', socket.id);
+  }
+  @ws.connect()
+  connect(@ws.socket() socket: Socket) {
+    console.log('Client connected: %s', socket.id);
+  }
+
+  @ws.subscribe('chat message')
+  handleChatMessage(msg: unknown, @ws.socket() socket: Socket) {
+    console.log('Chat message: %s', msg);
+    socket.nsp.emit('chat message', `[${socket.id}] ${msg}`);
```

Comment:
isn't?
```suggestion
    this.socket.nsp.emit('chat message', `[${socket.id}] ${msg}`);
```

Reply:
i think should be possible to response just returning a value

Reply:
The case of `handleChatMessage` method wants to show how inject the socket in the method param, but both cases work.

The response to the event is the method return value. That is explained below in the readme. The `socket.nsp.emit` want to show you how emit and event to a namespace or socket.